Client/Frontend/Browser/BackForwardListViewController.swift

```diff
@@ -4,56 +4,203 @@
+        let blurEffect = UIBlurEffect(style: isPrivate ? .Dark : .ExtraLight)
+        let blurEffectView = UIVisualEffectView(effect: blurEffect)
+        tableView.backgroundView = blurEffectView
+    }
+    
+    func loadSites(backForwardList: WKBackForwardList) {
+        let sql = profile.favicons as! SQLiteHistory
+        var urls: [String] = [String]()
+        
+        for page in backForwardList.forwardList.reverse() {
+            urls.append(page.URL.absoluteString)
+            listData.append((page, .Forward))
         }
-
-        listTableView.snp_makeConstraints { make in
-            make.top.equalTo(toolbar.snp_bottom)
+        
+        var currentRow = 0
+        if let currentPage = backForwardList.currentItem {
+            currentRow = listData.count
+            urls.append(currentPage.URL.absoluteString)
+            listData.append((currentPage, .Current))
+        }
+        for page in backForwardList.backList.reverse() {
+            urls.append(page.URL.absoluteString)
+            listData.append((page, .Backward))
+        }
+        
+        listData = listData.filter { !(($0.item.title ?? "").isEmpty)}
+        
+        sql.getSitesForURLs(urls).uponQueue(dispatch_get_main_queue()) { result in
+            if let cursor = result.successValue {
+                for cursorSite in cursor {
+                    if let site = cursorSite, let url = site?.url {
+                        self.sites[url] = site
+                    }
+                }
+                self.tableView.reloadData()
+            }
+        }
+        
+        if currentRow > 1 {
+            scrollTableViewToIndex(currentRow)
+        }
+    }
+    
+    func scrollTableViewToIndex(index: Int) {
+        let moveToIndexPath = NSIndexPath(forRow: index-2, inSection: 0)
+        self.tableView.reloadRowsAtIndexPaths([moveToIndexPath], withRowAnimation: .None)
+        self.tableView.scrollToRowAtIndexPath(moveToIndexPath, atScrollPosition: UITableViewScrollPosition.Middle, animated: false)
+    }
+    
+    override func didRotateFromInterfaceOrientation(fromInterfaceOrientation: UIInterfaceOrientation) {
+        self.resizeHeight()
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.addSubview(tableView)
+        tableView.snp_makeConstraints { make in
+            make.height.equalTo(0)
             make.left.right.bottom.equalTo(self.view)
         }
+        self.view.layoutIfNeeded()
+        self.resizeHeight()
+        
+        setupDismissTap()
     }
-
-    func SELdidClickDone() {
-        dismissViewControllerAnimated(true, completion: nil)
+    
+    func setupDismissTap() {
+        let tap = UITapGestureRecognizer(target: self, action: #selector(BackForwardListViewController.handleTap))
+        tap.cancelsTouchesInView = false
+        tap.delegate = self
+        parentViewController?.view.addGestureRecognizer(tap)
```

Comment:
Visually does that mean that the menu would appear on top of the back forward list? I feel like the menu and back forward list would be mutually exclusive, ie., if we're showing the back forward list, dismiss, then show the menu. @tecgirl would we want the menu to overlap on top of the back forward list?